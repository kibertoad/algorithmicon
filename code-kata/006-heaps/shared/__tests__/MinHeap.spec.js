const { MinHeap } = require('../MinHeap')

describe('MinHeap', () => {
  it('Correctly sorts elements', () => {
    const heap = new MinHeap()
    heap.insert(1)
    heap.insert(3)
    heap.insert(6)
    heap.insert(2)
    heap.insert(4)
    heap.insert(5)
    heap.insert(8)
    heap.insert(9)
    heap.insert(10)

    expect(heap.heap).toMatchSnapshot()
  })

  it('Correctly removes min element', () => {
    const heap = new MinHeap()
    heap.insert(1)
    heap.insert(3)
    heap.insert(6)
    heap.insert(2)
    heap.insert(4)
    heap.insert(5)
    heap.insert(8)
    heap.insert(9)
    heap.insert(10)

    const minValue = heap.extractMin()
    expect(minValue).toEqual(1)
    expect(heap.heap).toMatchSnapshot()
  })
})
