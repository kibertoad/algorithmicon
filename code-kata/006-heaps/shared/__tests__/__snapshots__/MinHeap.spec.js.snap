// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinHeap Correctly removes min element 1`] = `
Array [
  2,
  3,
  5,
  9,
  4,
  6,
  8,
  10,
]
`;

exports[`MinHeap Correctly sorts elements 1`] = `
Array [
  1,
  2,
  5,
  3,
  4,
  6,
  8,
  9,
  10,
]
`;
